src/classes/ALLO_Allocations_TEST.cls

```diff
@@ -1088,6 +1088,228 @@ private with sharing class ALLO_Allocations_TEST {
         System.assertEquals (null, allos[0].Percent__c, 'Allocation 0 - Percent');
     }
 
+    /**
+    * @description Tests that Allocations are updated and preserved when changing a Payment from a
+    * non zero to a zero amount, and that default allocation is removed
+    */
+    @isTest
+    private static void changePaymentAmountFromNonZeroToZeroshouldResizePercentAllocations() {
+        List<General_Accounting_Unit__c> gaus = new List<General_Accounting_Unit__c>();
+
+        General_Accounting_Unit__c defaultGau = new General_Accounting_Unit__c(Name='General');
```

Comment:
There is a `UTIL_UnitTestData_TEST.createGAUs()` method that you can use for these. Plus, all four of these new test methods have extremely similar data setup. Might be worth centralizing this is a helper method.

Reply:
I will see what I can do with this today/tomorrow (lot of meetings today).